.github/workflows/switch_js_branch.yaml

```diff
@@ -0,0 +1,45 @@
+name: switch branch
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+        description: The version of the release
+
+jobs:
+  branch:
+    runs-on: ubuntu-latest
+    env:
+      BRANCH: releases/${{ inputs.version }}
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: setup git config
+        run: |
+          git config user.name "GitHub Action"
+          git config user.email "github-action@users.noreply.github.com"
+
+      - name: install dependencies
+        run: |
+          npm install --no-bin-links
+
+      - name: build
+        run: npm run build
+
+      - name: switch or create branch
+        run: |
+            if git ls-remote --exit-code --heads origin ${BRANCH} ; then 
+                git checkout ${BRANCH}
+                git merge main
```

Comment:
Since we are using the `releases/nightly` branch only in the nightly release workflow and merging from main each time, I thought it'll avoid merge conflicts inherently.  I couldn't find a way to avoid conflicts. Will rebase help?

Reply:
Added code to avoid merge conflicts.